test/terra/backends/test_pulse_simulator.py

```diff
@@ -0,0 +1,302 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+PulseSimulator Integration Tests
+"""
+
+import unittest
+from test.terra import common
+
+import numpy as np
+
+import qiskit
+import qiskit.pulse as pulse
+
+from qiskit.compiler import assemble
+
+from qiskit.test.mock import FakeOpenPulse2Q
+from qiskit.pulse.commands import SamplePulse
+
+class TestPulseSimulator(common.QiskitAerTestCase):
+    """ PulseSimulator tests.
+
+    Uses Hamiltonian H = -0.5*omega0*sigmaz + omega1*os(omega*t+phi)*sigmax,
+    as it is analytically solvable. Action of the gates is computed by performing
+    the rotating frame transformation R = e^(-i*omega*t*sigmaz/2). The Hamiltonian in the
+    rotating frame, applying the Rotating Wave Approximation (RWA), is
+    Hrot = 0.5*(omega-omega0)*sigmaz + 0.5*omega1*cos(phi)*sigmax - 0.5*omega1*sin(phi)*sigmay
+    (note time independence). The unitary evolution is then given by Urot = e^(-i*Hrot*t).
+    """
+
+    def setUp(self):
+        """ Set configuration settings for pulse simulator """
+        qiskit.IBMQ.load_account()
+
+        #Get a pulse configuration from the fake backend
+        self.backend_fake = FakeOpenPulse2Q()
+        self.back_config = self.backend_fake.configuration()
+        self.back_config_dict = self.back_config.to_dict()
+        self.system = pulse.PulseChannelSpec.from_backend(self.backend_fake)
+        self.defaults = self.backend_fake.defaults()
+
+        self.qubit = 0 # use 0 qubit for tests
+        self.freq_qubit = self.defaults.qubit_freq_est[self.qubit]
+
+        #Get pulse simulator backend
+        self.backend_sim = qiskit.Aer.get_backend('pulse_simulator')
+
+        self.drive_samples = 100
+
+    def create_qobj(self, shots, omega0, omega1, omega, phi, meas_level):
```

Comment:
This test doesn't use all available pulse commands `Framechange` and `PersistentValue` is missing. We should also have tests for these.